onmt/transforms/bart.py

```diff
@@ -88,6 +88,12 @@ def __init__(self, vocab, mask_tok=DefaultTokens.MASK, mask_ratio=0.0,
         self.mask_length = mask_length
         self.poisson_lambda = poisson_lambda
 
+    @staticmethod
+    def set_random_seed(seed):
+        """Call this before use to ensure reproducibility."""
+        np.random.seed(seed)
+        torch.manual_seed(seed)
+
```

Comment:
In the unittest, we test on the `BARTNosing` module rather than the `BARTTransform` to avoid create a namespace contain all the parameters required by `opts`. As a random seed is needed to make the test reproducible, therefore I moved the related logic inside the `BARTNoising`.

Reply:
Right, missed this in the diffs.